### DISCUSSION TITLE
Updating the field value doesnt respect field type

### DISCUSSION THREAD

**Discussion by alucarddelta** [pydantic/pydantic#4703] [Question] [answered]
Is it just me or is this not the expected behavior.

I have a model with a set type a field. On creation new instance, it respects the field type rules, and raises errors accordingly.
However if I was to change the field value to what should be an incorrect value, it still allows it.

Shouldn't it still follow the field type and reject the updated value?

Python = `3.10`
Pydantic = `1.10.2`

```python
from pydantic import BaseModel

class foo(BaseModel):
    bar: bool

baz1 = foo(bar=True)  # Works as expected

baz1.bar = "TA-DA"  # Works when It shouldn't?!

baz2 = foo(bar="TA-DA")  # Fails as expected
```

I know there is a thread about bool strictness (eg.False == "False", "off", 0 ect) but this is clearly not related.

**Comment by samuelcolvin** [comment, marked as answer]
You want to add `validate_assignment` in config, see https://pydantic-docs.helpmanual.io/usage/model_config/#options

**Reply by alucarddelta** [reply]
That worked, Cheers.
